prog/minio/minio_cluster_nexus.rb

```diff
@@ -64,13 +72,30 @@ def before_run
   end
 
   label def wait
+    if !minio_cluster.root_certs.empty? && minio_cluster.certificate_last_checked_at < Time.now - 60 * 60 * 24 * 30 # ~1 month
+      hop_refresh_certificates
+    end
+
     when_reconfigure_set? do
       hop_reconfigure
     end
 
     nap 30
   end
 
+  label def refresh_certificates
+    if OpenSSL::X509::Certificate.new(minio_cluster.root_cert_1).not_after < Time.now + 60 * 60 * 24 * 30 * 5
+      minio_cluster.root_cert_1, minio_cluster.root_cert_key_1 = minio_cluster.root_cert_2, minio_cluster.root_cert_key_2
+      minio_cluster.root_cert_2, minio_cluster.root_cert_key_2 = Util.create_root_certificate(common_name: minio_cluster.ubid, duration: 60 * 60 * 24 * 365 * 10)
+      minio_cluster.servers.map(&:incr_reconfigure)
```

Comment:
Do we need to do restart? Below, when we increment reconfigure, we also increment restart.

Reply:
restarts in minio are not disruptive. They only take a second and only the existing connections are impacted. The restart is definitely needed to move a non tls enabled cluster to a tls enabled form since the connection strings change. I will look into this further and optimise the restart routine after the deployment of this pr.

Reply:
Got it. My concern is if we are in the middle of full base backup taking, which is a long operation, restart would cancel it.

It is OK to optimize the restart routine after this PR, but I think it is important enough to look soon without postponing much.